ambry-api/src/main/java/com.github.ambry/account/AccountBuilder.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.account;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import static com.github.ambry.account.Account.*;
+import static com.github.ambry.account.Container.*;
+
+
+/**
+ * A builder class for {@link Account}. Since {@link Account} is immutable, modifying an {@link Account} needs to
+ * build a new {@link Account} object with updated fields through this builder. An {@link Account} can be built
+ * in two ways: 1) from an existing {@link Account} object; and 2) by supplying required fields of an {@link Account}.
+ */
+public class AccountBuilder {
+  private Short id;
+  private String name;
+  private Account.AccountStatus status;
+  private short version;
+  private Map<Short, Container> idToContainerMetadataMap = new HashMap<>();
+
+  /**
+   * Constructor. This will allow building a new {@link Account} from an existing {@link Account}. The builder will
+   * include all the information including the {@link Container}s of the existing {@link Account}.
+   * @param origin The {@link Account} to build from.
+   */
+  public AccountBuilder(Account origin) {
+    if (origin == null) {
+      throw new IllegalArgumentException("origin cannot be null.");
+    }
+    this.id = origin.getId();
+    this.name = origin.getName();
+    this.status = origin.getStatus();
+    for (Container container : origin.getAllContainers()) {
+      idToContainerMetadataMap.put(container.getId(), container);
+    }
+    this.version = ACCOUNT_METADATA_VERSION_1;
```

Comment:
something to think about. Incase we introduce a new version for Account sometime in future, how will this code look like. Some fields in V2 may not be available in V1. Do we have dummy values for it ?